LeanKit-Labs/node-hilo
# Bump node & deps, add type definition

### Short description of the work completed

> This updates to node 12 (intentionally lower version to target a broader range of consumers), bumps dependencies, and adds a type definition file.

FYI- there are what error messages generated by `processhost` when running tests. Processhost is dev-only and the failures do not prevent the tests from passing, so I think it's safe to ignore them. If you see a way to prevent them altogether, let me know!

Because this introduces a new node version that may or may not be supported by a consumer, perhaps this should be a major version bump?

### Steps to test (if not obvious)

1. npm link in automation-service
2. remove `node-hilo` definition from types folder
3. verify types are loaded and working as expected
4. verify functionality in another js service

### For Reviewer Use Only

- [ ] Is the version upgrade path clear for this change? (breaking vs minor vs patch)
- [ ] Verify PR is being merged into an upstream feature branch. (NOT DEVELOP)
